pkg/stacks/unpack.go

```diff
@@ -108,6 +108,7 @@ type StackGroup struct {
 // StackPackager implentations can build a stack from Stack resources and emit the Yaml artifact
 type StackPackager interface {
 	SetApp(v1alpha1.AppMetadataSpec)
+	SetBehavior(unstructured.Unstructured)
```

Comment:
Without digging too deep into this, I wonder if we should use an interface instead of the unstructured.Unstructured type.

I realize SetInstall is doing the same thing, but I imagine the `Stackinatator` interface might be a better fit there.

You also seem keen to try something else here, per https://github.com/crossplaneio/crossplane/pull/1131/files#diff-8452f5d38b0987fceae61edde51d44ccR137

Reply:
True - it seems like we should be able to use an interface as the reference type, while using an `Unstructured` as the struct type.